### PR TITLE
added validation for register

### LINKED ISSUE #4: register validation
# Description 

If value isn't valid make borderColor red / otherwise green
# Suggestion
The input while the user registers should be tested for the same requirements like in the backend.

### PR DESCRIPTION
**Pull request by Akida31**
## Description

before you can register, your input will be checked. The user gets graphical output.
## Related Issue
closes #4 

## Type of change

- [ ] Bug fix
- [x] Feature request
- [ ] Documenting code

## Checklists

- [x] I referenced the according Issue
  - [x] I created an Issue
  - [x] I put a link in the Section [Related Issue](#related-issue)
- [ ] I tested my code with multiple browsers
- [x] My commit messages are good and understandable

### CODE REVIEW

File: src/components/Register/Register.scss [hunk 1]
```diff
@@ -19,6 +19,21 @@
...
+      margin-top: 0.8rem;
+      text-align: center;
+      width: 100%;
+    }
+
+    .success-text {
+      color: $color-green;
+      margin-top: 0.8rem;
+      text-align: center;
+      width: 100%;
```
Comment (msentwickler) [inline src/components/Register/Register.scss:33]: Duplicated code

File: src/components/Register/Register.tsx [hunk 2]
```diff
@@ -6,11 +6,37 @@ import TextInput from '../Input/TextBox/TextInput';
...
 import RegisterStore from '../../store/RegisterStore';
 import { t } from '../../i18n/i18n';
+import Validation from '../../lib/validation';
 
 const registerStore = new RegisterStore();
 
 @observer
 class Register extends React.Component {
+    registerHandler = () => {
+        let errorText = null;
```
Comment (msentwickler) [inline src/components/Register/Register.tsx:16]: :String?

File: src/css/colors.scss [hunk 3]
```diff
@@ -1,14 +1,15 @@
+$color-primary: #5bc0be;
```
Comment (msentwickler) [inline src/css/colors.scss:1]: Why?

File: src/store/RegisterStore.ts [hunk 4]
```diff
@@ -7,6 +7,8 @@ class RegisterStore {
     @observable isGtcAccepted = false;
     @observable accountType: 'teacher' | 'student' | 'unset' = 'unset';
     @observable teacherToken = '';
+    @observable errorText: string = null;
```
Comment (msentwickler) [inline src/store/RegisterStore.ts:10]: Why not '' instead of null

File: src/store/RegisterStore.ts [hunk 5]
```diff
@@ -26,6 +28,15 @@ class RegisterStore {
     @action setTeacherToken(value: string) {
         this.teacherToken = value;
     }
+    @action setErrorText(value: string) {
+        this.errorText = value;
+    }
+    @action clearErrorText() {
+        this.errorText = null;
```
Comment (msentwickler) [inline src/store/RegisterStore.ts:35]: ...

File: src/components/Register/Register.tsx (hunk 2 above)
Comment (Akida31) [reply src/components/Register/Register.tsx:16]: question mark does not work. Only `:string` possible

File: src/css/colors.scss (hunk 3 above)
Comment (Akida31) [reply src/css/colors.scss:1]: moved arround. Colors that are used more often should appear at the top

File: src/store/RegisterStore.ts [hunk 6]
```diff
@@ -7,6 +7,8 @@ class RegisterStore {
     @observable isGtcAccepted = false;
     @observable accountType: 'teacher' | 'student' | 'unset' = 'unset';
     @observable teacherToken = '';
+    @observable errorText: string = "";
```
Comment (msentwickler) [inline src/store/RegisterStore.ts:10]: single quotes

**Pull request merged**